### ISSUE TITLE
App data and file data still available unencrypted after locking Private Space

### ISSUE THREAD

**Issue by ryrona2** [GrapheneOS/os_issue_tracker#4305]
Expected behavior:

Locking the Private Space should unload the file encryption key in the same way that is done when shutting down a secondary user profile.

Actual behavior:

Contrary to user expectation, once the Private Space has been unlocked once, the app data and file data for the Private Space will remain available in unencrypted form until the next reboot of the phone. Locking the Private Space has no effect on the availability of the data inside the Private Space.

Steps to reproduce:

1. Factory default device and setup owner profile.
2. Create a Private Space.
3. Reboot phone.
4. Login to the owner profile.
5. Connect USB to computer and run "adb root" and then "adb shell".
6. In terminal, go to "/data/media/10" and do "ls -laR". All file names are encrypted. Try to "cat" a file. It will fail with a statement encryption key isn't loaded. Repeat for "/data/user/10".
7. Unlock the Private Space from the phone UI.
8. In terminal, go to "/data/media/10" and do "ls -laR". All file names are now fully readable. Try to "cat" a file. It will list the content successfully. Repeat for "/data/user/10".
9. Lock the Private Space again from the phone UI.
10. In terminal, go to "/data/media/10" and do "ls -laR". All file names are still fully readable. Try to "cat" a file. It will still list the content successfully. Repeat for "/data/user/10".

This issue is the same no matter in what manner the Private Space was locked (manually or automatically), or what credential modes is used for the Private Space (same or separate credentials, PIN or password, fingerprint or no fingerprint). Rebooting the device is the only way to unload the file encryption key for the Private Space.

**Comment by studog-github** [comment]
I'm wondering if what you're seeing is an effect of filesystem caching. In Step 10, are you `cat`ing a different file that in Step 8?

**Comment by ryrona2** [comment]
> I'm wondering if what you're seeing is an effect of filesystem caching. In Step 10, are you `cat`ing a different file that in Step 8?

Yes, I specifically did that, since I did observe some caching behavior like what you say for secondary user profiles. But for the Private Space the encryption key definitely remains loaded and I can read any file, cached or not.

**Comment by Dark-Aeon** [comment]
update on the issue on my Phone:

today again the quick tiles dont react with around 3minutes delay.

I checked under Accessibility  and KDE connect and Bitwarden show as off. 

Just reporting it as another data point.

**Comment by Dark-Aeon** [comment]
I did. I apologize.

**Comment by rusty-snake** [comment]
> future new release will add it back with a toggle.

Toggle is now there. Anything left?
